File: examples/javascript/basic-usage.js
/**
 * AFXO JavaScript SDK - Basic Usage Examples
 *
 * Install: npm install @afxo/sdk
 */

// ============ REST API Examples ============

// Using fetch (no SDK required)
async function getRate(base, quote, apiKey) {
  const response = await fetch(
    `https://api.afxo.ai/v1/rates/${base}/${quote}`,
    {
      headers: {
        'X-API-Key': apiKey,
      },
    }
  );

  if (!response.ok) {
    throw new Error(`API error: ${response.status}`);
  }

  return response.json();
}

// Example usage
async function main() {
  const apiKey = process.env.AFXO_API_KEY;

  // Get KES/USD rate
  const kesRate = await getRate('KES', 'USD', apiKey);
  console.log('KES/USD Rate:', kesRate.rate);
  console.log('Confidence:', kesRate.confidence);
  console.log('Sources:', kesRate.sources);

  // Get multiple rates
  const pairs = ['KES/USD', 'NGN/USD', 'GHS/USD'];
  const response = await fetch(
    `https://api.afxo.ai/v1/rates/batch?pairs=${pairs.join(',')}`,
    {
      headers: { 'X-API-Key': apiKey },
    }
  );
  const batchRates = await response.json();

  for (const rate of batchRates.rates) {
    console.log(`${rate.pair}: ${rate.rate} (confidence: ${rate.confidence})`);
  }
}

main().catch(console.error);

// ============ WebSocket Example ============

function subscribeToRates(apiKey, pairs, onUpdate) {
  const ws = new WebSocket('wss://api.afxo.ai/v1/ws');

  ws.onopen = () => {
    // Authenticate
    ws.send(JSON.stringify({ type: 'auth', apiKey }));

    // Subscribe to pairs
    ws.send(JSON.stringify({ type: 'subscribe', pairs }));
  };

  ws.onmessage = (event) => {
    const data = JSON.parse(event.data);

    switch (data.type) {
      case 'rate':
        onUpdate(data);
        break;
      case 'error':
        console.error('WebSocket error:', data.message);
        break;
      case 'heartbeat':
        // Keep-alive, no action needed
        break;
    }
  };

  ws.onerror = (error) => {
    console.error('WebSocket error:', error);
  };

  return ws;
}

// WebSocket usage
// const ws = subscribeToRates(
//   process.env.AFXO_API_KEY,
//   ['KES/USD', 'NGN/USD'],
//   (update) => {
//     console.log(`${update.pair} updated: ${update.rate}`);
//   }
// );

File: examples/python/basic_usage.py
"""
AFXO Python Examples - Basic Usage

Install requests: pip install requests
"""

import os
import requests
from typing import Optional

BASE_URL = "https://api.afxo.ai/v1"


class AFXOClient:
    """Simple AFXO API client."""

    def __init__(self, api_key: str):
        self.api_key = api_key
        self.session = requests.Session()
        self.session.headers.update({"X-API-Key": api_key})

    def get_rate(self, base: str, quote: str) -> dict:
        """Get current exchange rate for a currency pair."""
        response = self.session.get(f"{BASE_URL}/rates/{base}/{quote}")
        response.raise_for_status()
        return response.json()

    def get_batch_rates(self, pairs: list[str]) -> dict:
        """Get rates for multiple currency pairs."""
        pairs_param = ",".join(pairs)
        response = self.session.get(f"{BASE_URL}/rates/batch?pairs={pairs_param}")
        response.raise_for_status()
        return response.json()

    def get_history(
        self,
        base: str,
        quote: str,
        from_date: str,
        to_date: str,
        interval: str = "daily",
    ) -> dict:
        """Get historical rate data."""
        response = self.session.get(
            f"{BASE_URL}/rates/{base}/{quote}/history",
            params={"from": from_date, "to": to_date, "interval": interval},
        )
        response.raise_for_status()
        return response.json()

    def get_status(self) -> dict:
        """Get network status."""
        response = self.session.get(f"{BASE_URL}/status")
        response.raise_for_status()
        return response.json()


def main():
    # Get API key from environment
    api_key = os.environ.get("AFXO_API_KEY")
    if not api_key:
        print("Please set AFXO_API_KEY environment variable")
        return

    client = AFXOClient(api_key)

    # Get single rate
    print("=== KES/USD Rate ===")
    kes_rate = client.get_rate("KES", "USD")
    print(f"Rate: {kes_rate['rate']}")
    print(f"Confidence: {kes_rate['confidence']}%")
    print(f"Sources: {kes_rate['sources']}")
    print()

    # Get batch rates
    print("=== Batch Rates ===")
    batch = client.get_batch_rates(["KES/USD", "NGN/USD", "GHS/USD", "ZAR/USD"])
    for rate in batch["rates"]:
        print(f"{rate['pair']}: {rate['rate']} (confidence: {rate['confidence']}%)")
    print()

    # Check network status
    print("=== Network Status ===")
    status = client.get_status()
    print(f"Status: {status['status']}")
    print(f"Operators: {status['operators']}")
    print(f"Currencies: {status['currencies']}")


if __name__ == "__main__":
    main()
